hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -950,10 +141,41 @@ public boolean shouldRun() {
   /**
    * Checks if ContainerBalancer is currently running in this SCM.
    *
-   * @return true if the currentBalancingThread is not null, otherwise false
+   * @return true if balancer started, otherwise false
    */
   public boolean isBalancerRunning() {
-    return currentBalancingThread != null;
+    return (null != task
+        && task.getBalancerStatus() == ContainerBalancerTask.Status.RUNNING);
+  }
+
+  /**
+   * Checks if ContainerBalancer in valid state and can be started.
+   *
+   * @return true if balancer can be started, otherwise false
+   */
+  private boolean canBalancerStart() {
+    return (null == task
+        || task.getBalancerStatus() == ContainerBalancerTask.Status.STOPPED);
+  }
+
+  /**
+   * get the Container Balancer state.
+   *
+   * @return true if balancer started, otherwise false
+   */
+  public ContainerBalancerTask.Status getBalancerStatus() {
+    return null != task ? task.getBalancerStatus()
+        : ContainerBalancerTask.Status.STOPPED;
+  }
+
+  /**
+   * Checks if ContainerBalancer is in valid state to call stop.
+   *
+   * @return true if balancer can be stopped, otherwise false
+   */
+  private boolean canBalancerStop() {
+    return null != task
+        && task.getBalancerStatus() == ContainerBalancerTask.Status.RUNNING;
   }
```

Comment:
This method seems redundant. `isBalancerRunning` can be used instead. If we want to have `canBalancerStop` for better readability, let's just call `isBalancerRunning` inside it?

Reply:
There is a state machine,

Reply:
1. save configuration is done when stop is done by balancerTask, as that can tell actually its stopped
2. isBalancerRunning just check if Running state. But canBalancerStop also check if stopping and disallow that state also if stop is called again.